lib/common/api.dart

```
@@ -0,0 +1,168 @@
+import 'package:goswapinfo/common/pair_liquidity.dart';
+import 'package:goswapinfo/common/token.dart';
+import 'package:goswapinfo/common/volume.dart';
+import 'package:goswapinfo/common/pair_volume.dart';
+import 'package:http/http.dart' as http;
+import 'dart:convert';
+import 'total.dart';
+
+class Api {
+  static final String apiUrl = 'https://goswap-stats-xcefncm5jq-uc.a.run.app';
+  static Future<List<Total>> fetchTotals() async {
+    String url = apiUrl + '/totals';
+    http.Response response;
+    try {
+      response = await http.get(url);
+    } catch (err) {
+      throw err;
+    }
+    if (response.statusCode != 200) {
+      if (response.statusCode == 404) return Future.value(null);
```

Maybe these should return an error?  Otherwise a progress indicator may just spin forever.

---

fixed